web/src/hashchange.js

```diff
@@ -145,7 +145,13 @@ export function set_hash_to_default_view() {
     }
 }
 
+function hide_non_message_list_views() {
+    maybe_hide_inbox();
+    maybe_hide_recent_view();
+}
+
 function show_default_view() {
+    hide_non_message_list_views();
```

Comment:
This part seems a bit weird; the existing `show` methods for inbox and recent view have logic like this:

```
    // Hide "middle-column" which has html for rendering                                                
    // a messages narrow. We hide it and show Recent Conversations.                                     
    $("#message_feed_container").hide();                                                                
    $("#recent_view").show();                                                                           
    set_visible(true);                                                                                  
```

And given there's other code paths for showing those, I suspect the bug might be in those functions instead?

Reply:
Maybe it's cleaner to fix this with #26688. It's subjective where the bug actually lies though.

Reply:
The problem is calling show_default_view without hiding other views. Like showing recent view without hiding inbox view.

Reply:
```js
        case "":
        case "#":
            show_default_view();
            break;
        case "#recent":
            maybe_hide_inbox();
            recent_view_ui.show();
            break;
        case "#inbox":
            maybe_hide_recent_view();
            inbox_ui.show();
            break;
```

This change is basically to handle the first case here. Other cases already hide other views.